# Capstone Project - The Battle of Neighborhoods (Week 1)

## Part 1 [Week 1]



Clearly define a problem or an idea of your choice, where you would need to leverage the Foursquare location data to solve or execute. Remember that data science problems always target an audience and are meant to help a group of stakeholders solve a problem, so make sure that you explicitly describe your audience and why they would care about your problem.

## Part 2 [Week 1]

Describe the data that you will be using to solve the problem or execute your idea. Remember that you will need to use the Foursquare location data to solve the problem or execute your idea. You can absolutely use other datasets in combination with the Foursquare location data. So make sure that you provide adequate explanation and discussion, with examples, of the data that you will be using, even if it is only Foursquare location data.


## Part 3 [Week 2]

A full report consisting of all of the following components (15 marks):
    
- Introduction where you discuss the business problem and who would be interested in this project.
- Data where you describe the data that will be used to solve the problem and the source of the data.
- Methodology section which represents the main component of the report where you discuss and describe any exploratory data analysis that you did, any inferential statistical testing that you performed, and what machine learnings were used and why.
- Results section where you discuss the results.
- Discussion section where you discuss any observations you noted and any recommendations you can make based on the results.
- Conclusion section where you conclude the report.
- A link to your Notebook on your Github repository pushed showing your code. (15 marks).
- Your choice of a presentation or blogpost. (10 marks)

## Section 1: Introduction 

### Background

The London Housing Market is on a decline and facing several economic challenges. There  is a chance of higher taxes and the Bank fo England is warning home owners that property values can decrease up to 30% in the case of an exit from the European Union. Four concerning signs of hidden price falls, record low sales, exodus from homebuilder, and increaese in taxes in the London Housing market offers hints that should concern citizens in England and Wales. 


### Project Idea

To help the home buyers in the London House Markek, we can use machine learning to help them anlayze and make better decisions for buying homes. Thus, we can ask the questiion: How can we help these buyers purchase a proprerty in spite of all the financial and economic challenges? To answer this question, we are going to use the cluster method on London Neighborhoods to find average price of houses, and recommend them to pruchase a real estate that will net them back a positive value investment. We will based our "postive value" based on proprty amenitites, schools, groceery stores, venues, shopping mall, and entertainment factors. 

## Section 2: Data

Our source for London properties were taken from http://landregistry.data.gov.uk/. We will use and analyze the data from Price Paid Data the following fields: Postcode, PAON (Primary Addressable obeject Name), SAON (Secondary Addressable Object Name.


To analyze and recommmend locations across different venues of "postive" values, we will use FourSquare API to access data and present it as a dataframe in form of data visualization. We can combine the data on London proporties with the relative price paid data from the data souce link and data of "positive" values near properties from the FourSquare API interface. This will allow us to recommend to buyers of houses with postivie value investment. 

## Section 3: Methodology

This section will divide the main parts of the analysis into four main parts:

- Collection and Inspection of Data
- Exploring and Understanding Data
- Data Preparation/Preprocessing
- Modeling

We then can make our predictions based on our modeling

**i. Collection and Inspection of Data**

In [1]:
import os # Operating System
import numpy as np
import pandas as pd
import datetime as dt # Datetime
import json # library to handle JSON files

!conda install -c conda-forge geopy --yes
from geopy.geocoders import Nominatim # convert an address into latitude and longitude values

import requests # library to handle requests
from pandas.io.json import json_normalize # tranform JSON file into a pandas dataframe

# Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors

!conda install -c conda-forge folium=0.5.0 --yes
import folium #import folium # map rendering library

print('Libraries imported.')

Solving environment: done

# All requested packages already installed.

Solving environment: done

# All requested packages already installed.

Libraries imported.


In [2]:
#Read the data for examination (Source: http://landregistry.data.gov.uk/)
df_ppd = pd.read_csv("http://prod2.publicdata.landregistry.gov.uk.s3-website-eu-west-1.amazonaws.com/pp-2018.csv")

**ii. Exploring and Understanding Data**

In [3]:
df_ppd.head(11)


{666758D7-43A9-3363-E053-6B04A8C0D74E}  405000  2018-01-25 00:00  \
0   {666758D7-43AA-3363-E053-6B04A8C0D74E}  315000  2018-01-23 00:00   
1   {666758D7-43AD-3363-E053-6B04A8C0D74E}  165000  2018-01-19 00:00   
2   {666758D7-43B0-3363-E053-6B04A8C0D74E}  370000  2018-01-22 00:00   
3   {666758D7-43B3-3363-E053-6B04A8C0D74E}  320000  2018-01-19 00:00   
4   {666758D7-43B4-3363-E053-6B04A8C0D74E}  180000  2018-01-31 00:00   
5   {666758D7-43B6-3363-E053-6B04A8C0D74E}  205000  2018-01-26 00:00   
6   {666758D7-43B7-3363-E053-6B04A8C0D74E}  255000  2018-01-26 00:00   
7   {666758D7-43B8-3363-E053-6B04A8C0D74E}  177500  2018-01-31 00:00   
8   {666758D7-43B9-3363-E053-6B04A8C0D74E}  174995  2018-02-02 00:00   
9   {666758D7-43BB-3363-E053-6B04A8C0D74E}  194950  2018-02-02 00:00   
10  {666758D7-43BC-3363-E053-6B04A8C0D74E}  300000  2018-02-06 00:00   

    WR15 8LH  D  N  F RAMBLERS WAY Unnamed: 8        Unnamed: 9  \
0    SY7 8QA  D  N  F  MONT CENISE        NaN               NaN   
1    SY1 2BF  T  Y  F           42        NaN        PENSON WAY   
2    SY8 4DF  D  N  F   WILLOW HEY        NaN               NaN   
3   TF10 7ET  D  N  F            3        NaN  PRINCESS GARDENS   
4    SY3 0NQ  S  N  F           79        NaN     LYTHWOOD ROAD   
5    SY4 5ES  D  N  F            7        NaN          WELLGATE   
6   TF11 9LS  D  N  F     MONTROSE        NaN       GROOMS LANE   
7    SY1 3DG  S  N  F           27        NaN       POWIS DRIVE   
8    TF1 5GN  D  N  F           89        NaN      HENDY AVENUE   
9    SY2 5SZ  S  Y  F           80        NaN    REDWING FIELDS   
10   SY3 8HT  D  N  F           31        NaN  WOODFIELD AVENUE   

            BORASTON TENBURY WELLS  SHROPSHIRE SHROPSHIRE.1  A A.1  
0               CLUN   CRAVEN ARMS  SHROPSHIRE   SHROPSHIRE  A   A  
1                NaN    SHREWSBURY  SHROPSHIRE   SHROPSHIRE  A   A  
2   ASHFORD CARBONEL        LUDLOW  SHROPSHIRE   SHROPSHIRE  A   A  
3                NaN       NEWPORT      WREKIN       WREKIN  A   A  
4       BAYSTON HILL    SHREWSBURY  SHROPSHIRE   SHROPSHIRE  A   A  
5                WEM    SHREWSBURY  SHROPSHIRE   SHROPSHIRE  A   A  
6          KEMBERTON       SHIFNAL  SHROPSHIRE   SHROPSHIRE  A   A  
7                NaN    SHREWSBURY  SHROPSHIRE   SHROPSHIRE  A   A  
8             KETLEY       TELFORD      WREKIN       WREKIN  A   A  
9                NaN    SHREWSBURY  SHROPSHIRE   SHROPSHIRE  A   A  
10               NaN    SHREWSBURY  SHROPSHIRE   SHROPSHIRE  A   A

In [4]:
df_ppd.shape

(1021214, 16)

The dataset has over 1,021,214 rows and 16 columns. 

**iii. Data Preparation/Preprocessing**

We will prepare the dataset for the modeling process using the following steps:

In [20]:
# 1. Rename and asssign column names to better understand the data
df_ppd.columns = ['TUID', 'Price', 'Date_Transfer', 'Postcode', 'Prop_Type', 'Old_New', 'Duration', 'PAON', \
                  'SAON', 'Street', 'Locality', 'Town_City', 'District', 'County', 'PPD_Cat_Type', 'Record_Status']

In [21]:
#2. Format the date column
df_ppd['Date_Transfer'] = df_ppd['Date_Transfer'].apply(pd.to_datetime)

#3. Delete all obsolete transactions which were done before 2016
df_ppd.drop(df_ppd[df_ppd.Date_Transfer.dt.year < 2016].index, inplace=True)

#4. Sort by Date of Sale
df_ppd.sort_values(by=['Date_Transfer'],ascending=[False],inplace=True)

In [22]:
df_ppd_london = df_ppd.query("Town_City == 'LONDON'")

#5.  Make a list of street names in LONDON
streets = df_ppd_london['Street'].unique().tolist()

In [23]:
df_grp_price = df_ppd_london.groupby(['Street'])['Price'].mean().reset_index()

#6. Give meaningful names to the columns
df_grp_price.columns = ['Street', 'Avg_Price']

In [9]:
#7. Input your Budget's Upper Limit and Lower Limit - Find the locations df_grp_price which fits your budget
df_affordable = df_grp_price.query("(Avg_Price >= 2200000) & (Avg_Price <= 2500000)")

In [10]:
#8. Display the dataframe
df_affordable

Street     Avg_Price
196            ALBION SQUARE  2.450000e+06
391              ANHALT ROAD  2.435000e+06
406          ANSDELL TERRACE  2.250000e+06
421          APPLEGARTH ROAD  2.400000e+06
699         AYLESTONE AVENUE  2.286667e+06
853          BARONSMEAD ROAD  2.375000e+06
979           BEAUCLERC ROAD  2.480000e+06
1100         BELVEDERE DRIVE  2.340000e+06
1213       BICKENHALL STREET  2.208500e+06
1251       BIRCHLANDS AVENUE  2.217000e+06
1550          BRAMPTON GROVE  2.456875e+06
1629       BRIARDALE GARDENS  2.397132e+06
1794                BROOKWAY  2.400000e+06
1911            BURBAGE ROAD  2.445000e+06
1977               BURY WALK  2.492500e+06
2064         CALLCOTT STREET  2.375000e+06
2125       CAMPDEN HILL ROAD  2.379653e+06
2132            CAMPION ROAD  2.461000e+06
2155           CANNING PLACE  2.425000e+06
2222           CARLISLE ROAD  2.200000e+06
2227         CARLTON GARDENS  2.483500e+06
2239           CARLYLE COURT  2.300000e+06
2402          CHALCOT SQUARE  2.286679e+06
2480            CHARLES LANE  2.414000e+06
2558        CHELSEA CRESCENT  2.495000e+06
2602     CHESTER CLOSE NORTH  2.450000e+06
2633            CHEYNE COURT  2.250000e+06
2636              CHEYNE ROW  2.410000e+06
2681           CHISWICK MALL  2.287500e+06
2756               CITY ROAD  2.468340e+06
...                      ...           ...
10896   RUSSELL GARDENS MEWS  2.300000e+06
11143         SETTLES STREET  2.487500e+06
11218         SHELDON AVENUE  2.349542e+06
11451          SOUTH END ROW  2.470000e+06
11529    SOUTHWOOD LAWN ROAD  2.350000e+06
11532         SOVEREIGN PARK  2.500000e+06
11749  ST MARGARETS CRESCENT  2.216500e+06
11785       ST OSWALDS PLACE  2.250000e+06
11805       ST PETERS SQUARE  2.468730e+06
11842       STAFFORD TERRACE  2.355000e+06
12180       SUTHERLAND PLACE  2.456000e+06
12242          SYDNEY STREET  2.240833e+06
12384            THAMES BANK  2.400000e+06
12446            THE HEXAGON  2.335000e+06
12725        TREDEGAR SQUARE  2.436666e+06
12787         TRINITY STREET  2.317500e+06
12937   UPPER HAMPSTEAD WALK  2.500000e+06
13198        WALPOLE GARDENS  2.303500e+06
13200         WALPOLE STREET  2.242500e+06
13274         WARWICK SQUARE  2.432273e+06
13371            WELBECK WAY  2.267000e+06
13383      WELLESLEY TERRACE  2.410000e+06
13394      WELLINGTON STREET  2.293155e+06
13526     WESTMORELAND PLACE  2.300000e+06
13633       WHITFIELD STREET  2.451000e+06
13678         WILFRED STREET  2.410538e+06
13704     WILLOW BRIDGE ROAD  2.425000e+06
13724          WILSON STREET  2.257500e+06
13752        WINCHENDON ROAD  2.350000e+06
13789           WINGATE ROAD  2.206400e+06

[160 rows x 2 columns]

In [11]:
import pandas as pd
import numpy as np
import datetime as DT
import hmac
from geopy.geocoders import Nominatim
from geopy.distance import vincenty
# import k-means from clustering stage
from sklearn.cluster import KMeans

In [12]:
for index, item in df_affordable.iterrows():
    print(f"index: {index}")
    print(f"item: {item}")
    print(f"item.Street only: {item.Street}")

index: 196
item: Street       ALBION SQUARE
Avg_Price         2.45e+06
Name: 196, dtype: object
item.Street only: ALBION SQUARE
index: 391
item: Street       ANHALT ROAD
Avg_Price      2.435e+06
Name: 391, dtype: object
item.Street only: ANHALT ROAD
index: 406
item: Street       ANSDELL TERRACE
Avg_Price           2.25e+06
Name: 406, dtype: object
item.Street only: ANSDELL TERRACE
index: 421
item: Street       APPLEGARTH ROAD
Avg_Price            2.4e+06
Name: 421, dtype: object
item.Street only: APPLEGARTH ROAD
index: 699
item: Street       AYLESTONE AVENUE
Avg_Price         2.28667e+06
Name: 699, dtype: object
item.Street only: AYLESTONE AVENUE
index: 853
item: Street       BARONSMEAD ROAD
Avg_Price          2.375e+06
Name: 853, dtype: object
item.Street only: BARONSMEAD ROAD
index: 979
item: Street       BEAUCLERC ROAD
Avg_Price          2.48e+06
Name: 979, dtype: object
item.Street only: BEAUCLERC ROAD
index: 1100
item: Street       BELVEDERE DRIVE
Avg_Price           2.34e+06
Name

In [13]:
geolocator = Nominatim()


/opt/conda/envs/Python36/lib/python3.6/site-packages/ipykernel/__main__.py:1: DeprecationWarning: Using Nominatim with the default "geopy/1.20.0" `user_agent` is strongly discouraged, as it violates Nominatim's ToS https://operations.osmfoundation.org/policies/nominatim/ and may possibly cause 403 and 429 HTTP errors. Please specify a custom `user_agent` with `Nominatim(user_agent="my-application")` or by overriding the default `user_agent`: `geopy.geocoders.options.default_user_agent = "my-application"`. In geopy 2.0 this will become an exception.
  if __name__ == '__main__':


In [14]:
df_affordable['city_coord'] = df_affordable['Street'].apply(geolocator.geocode).apply(lambda x: (x.latitude, x.longitude))

/opt/conda/envs/Python36/lib/python3.6/site-packages/ipykernel/__main__.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  if __name__ == '__main__':


In [15]:
df_affordable

Street     Avg_Price                         city_coord
196            ALBION SQUARE  2.450000e+06   (-41.27375755, 173.289393239104)
391              ANHALT ROAD  2.435000e+06           (51.4803265, -0.1667607)
406          ANSDELL TERRACE  2.250000e+06           (51.4998899, -0.1891027)
421          APPLEGARTH ROAD  2.400000e+06              (53.749244, -0.32678)
699         AYLESTONE AVENUE  2.286667e+06           (51.5409157, -0.2178742)
853          BARONSMEAD ROAD  2.375000e+06            (51.4773147, -0.239457)
979           BEAUCLERC ROAD  2.480000e+06           (51.4995771, -0.2290331)
1100         BELVEDERE DRIVE  2.340000e+06           (51.4249173, -0.2120774)
1213       BICKENHALL STREET  2.208500e+06           (51.5211969, -0.1589341)
1251       BIRCHLANDS AVENUE  2.217000e+06           (51.4483941, -0.1604676)
1550          BRAMPTON GROVE  2.456875e+06           (51.5703648, -0.2833944)
1629       BRIARDALE GARDENS  2.397132e+06           (51.5601748, -0.1954305)
1794                BROOKWAY  2.400000e+06    (45.4321849, -122.802811661158)
1911            BURBAGE ROAD  2.445000e+06           (51.4509669, -0.0952879)
1977               BURY WALK  2.492500e+06           (52.1455294, -0.4235933)
2064         CALLCOTT STREET  2.375000e+06           (51.5083499, -0.1983276)
2125       CAMPDEN HILL ROAD  2.379653e+06           (51.5064605, -0.1988955)
2132            CAMPION ROAD  2.461000e+06            (52.6813505, 0.9654391)
2155           CANNING PLACE  2.425000e+06             (53.4022012, -2.98726)
2222           CARLISLE ROAD  2.200000e+06          (42.5490988, -71.4166686)
2227         CARLTON GARDENS  2.483500e+06   (-37.80194335, 144.971970171017)
2239           CARLYLE COURT  2.300000e+06           (41.119072, -75.0615162)
2402          CHALCOT SQUARE  2.286679e+06           (51.5414545, -0.1552649)
2480            CHARLES LANE  2.414000e+06           (40.2375254, -83.349229)
2558        CHELSEA CRESCENT  2.495000e+06            (34.524738, -85.448539)
2602     CHESTER CLOSE NORTH  2.450000e+06           (51.5292054, -0.1450813)
2633            CHEYNE COURT  2.250000e+06             (51.599677, 0.5256231)
2636              CHEYNE ROW  2.410000e+06            (51.4837173, -0.169603)
2681           CHISWICK MALL  2.287500e+06           (51.4879942, -0.2466045)
2756               CITY ROAD  2.468340e+06           (51.5283061, -0.0919461)
...                      ...           ...                                ...
10896   RUSSELL GARDENS MEWS  2.300000e+06           (51.4998689, -0.2118904)
11143         SETTLES STREET  2.487500e+06           (51.5153237, -0.0643266)
11218         SHELDON AVENUE  2.349542e+06            (43.599287, -73.045426)
11451          SOUTH END ROW  2.470000e+06           (51.4987463, -0.1890787)
11529    SOUTHWOOD LAWN ROAD  2.350000e+06            (51.5746267, -0.146238)
11532         SOVEREIGN PARK  2.500000e+06    (-27.4937089, 153.210145350143)
11749  ST MARGARETS CRESCENT  2.216500e+06            (53.6074918, -0.267971)
11785       ST OSWALDS PLACE  2.250000e+06           (51.4872071, -0.1185341)
11805       ST PETERS SQUARE  2.468730e+06     (41.9022353, 12.4578392797771)
11842       STAFFORD TERRACE  2.355000e+06           (51.5009379, -0.1960492)
12180       SUTHERLAND PLACE  2.456000e+06           (54.7731795, -1.5438161)
12242          SYDNEY STREET  2.240833e+06         (-33.8051445, 151.1933273)
12384            THAMES BANK  2.400000e+06            (51.471715, -0.2689072)
12446            THE HEXAGON  2.335000e+06          (-31.8956726, 26.8758122)
12725        TREDEGAR SQUARE  2.436666e+06  (51.5270935, -0.0321842332596606)
12787         TRINITY STREET  2.317500e+06             (52.206141, 0.1181326)
12937   UPPER HAMPSTEAD WALK  2.500000e+06            (51.558467, -0.1774529)
13198        WALPOLE GARDENS  2.303500e+06           (51.4397435, -0.3412634)
13200         WALPOLE STREET  2.242500e+06             (52.626946, 1.2846547)
13274         WARWICK SQUARE  2.432273e+06  

In [16]:
df_affordable[['Latitude', 'Longitude']] = df_affordable['city_coord'].apply(pd.Series)

/opt/conda/envs/Python36/lib/python3.6/site-packages/pandas/core/frame.py:3391: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self[k1] = value[k2]


In [17]:
df_affordable

Street     Avg_Price                         city_coord  \
196            ALBION SQUARE  2.450000e+06   (-41.27375755, 173.289393239104)   
391              ANHALT ROAD  2.435000e+06           (51.4803265, -0.1667607)   
406          ANSDELL TERRACE  2.250000e+06           (51.4998899, -0.1891027)   
421          APPLEGARTH ROAD  2.400000e+06              (53.749244, -0.32678)   
699         AYLESTONE AVENUE  2.286667e+06           (51.5409157, -0.2178742)   
853          BARONSMEAD ROAD  2.375000e+06            (51.4773147, -0.239457)   
979           BEAUCLERC ROAD  2.480000e+06           (51.4995771, -0.2290331)   
1100         BELVEDERE DRIVE  2.340000e+06           (51.4249173, -0.2120774)   
1213       BICKENHALL STREET  2.208500e+06           (51.5211969, -0.1589341)   
1251       BIRCHLANDS AVENUE  2.217000e+06           (51.4483941, -0.1604676)   
1550          BRAMPTON GROVE  2.456875e+06           (51.5703648, -0.2833944)   
1629       BRIARDALE GARDENS  2.397132e+06           (51.5601748, -0.1954305)   
1794                BROOKWAY  2.400000e+06    (45.4321849, -122.802811661158)   
1911            BURBAGE ROAD  2.445000e+06           (51.4509669, -0.0952879)   
1977               BURY WALK  2.492500e+06           (52.1455294, -0.4235933)   
2064         CALLCOTT STREET  2.375000e+06           (51.5083499, -0.1983276)   
2125       CAMPDEN HILL ROAD  2.379653e+06           (51.5064605, -0.1988955)   
2132            CAMPION ROAD  2.461000e+06            (52.6813505, 0.9654391)   
2155           CANNING PLACE  2.425000e+06             (53.4022012, -2.98726)   
2222           CARLISLE ROAD  2.200000e+06          (42.5490988, -71.4166686)   
2227         CARLTON GARDENS  2.483500e+06   (-37.80194335, 144.971970171017)   
2239           CARLYLE COURT  2.300000e+06           (41.119072, -75.0615162)   
2402          CHALCOT SQUARE  2.286679e+06           (51.5414545, -0.1552649)   
2480            CHARLES LANE  2.414000e+06           (40.2375254, -83.349229)   
2558        CHELSEA CRESCENT  2.495000e+06            (34.524738, -85.448539)   
2602     CHESTER CLOSE NORTH  2.450000e+06           (51.5292054, -0.1450813)   
2633            CHEYNE COURT  2.250000e+06             (51.599677, 0.5256231)   
2636              CHEYNE ROW  2.410000e+06            (51.4837173, -0.169603)   
2681           CHISWICK MALL  2.287500e+06           (51.4879942, -0.2466045)   
2756               CITY ROAD  2.468340e+06           (51.5283061, -0.0919461)   
...                      ...           ...                                ...   
10896   RUSSELL GARDENS MEWS  2.300000e+06           (51.4998689, -0.2118904)   
11143         SETTLES STREET  2.487500e+06           (51.5153237, -0.0643266)   
11218         SHELDON AVENUE  2.349542e+06            (43.599287, -73.045426)   
11451          SOUTH END ROW  2.470000e+06           (51.4987463, -0.1890787)   
11529    SOUTHWOOD LAWN ROAD  2.350000e+06            (51.5746267, -0.146238)   
11532         SOVEREIGN PARK  2.500000e+06    (-27.4937089, 153.210145350143)   
11749  ST MARGARETS CRESCENT  2.216500e+06            (53.6074918, -0.267971)   
11785       ST OSWALDS PLACE  2.250000e+06           (51.4872071, -0.1185341)   
11805       ST PETERS SQUARE  2.468730e+06     (41.9022353, 12.4578392797771)   
11842       STAFFORD TERRACE  2.355000e+06           (51.5009379, -0.1960492)   
12180       SUTHERLAND PLACE  2.456000e+06           (54.7731795, -1.5438161)   
12242          SYDNEY STREET  2.240833e+06         (-33.8051445, 151.1933273)   
12384            THAMES BANK  2.400000e+06            (51.471715, -0.2689072)   
12446            THE HEXAGON  2.335000e+06          (-31.8956726, 26.8758122)   
12725        TREDEGAR SQUARE  2.436666e+06  (51.5270935, -0.0321842332596606)   
12787         TRINITY STREET  2.317500e+06             (52.206141, 0.1181326)   
12937   UPPER HAMPSTEAD WALK  2.500000e+06            (51.558467, -0.1774529)   
13198        WALPOLE GARDENS  2.303500e+06           

In [18]:
df = df_affordable.drop(columns=['city_coord'])


In [19]:
df


Street     Avg_Price   Latitude   Longitude
196            ALBION SQUARE  2.450000e+06 -41.273758  173.289393
391              ANHALT ROAD  2.435000e+06  51.480326   -0.166761
406          ANSDELL TERRACE  2.250000e+06  51.499890   -0.189103
421          APPLEGARTH ROAD  2.400000e+06  53.749244   -0.326780
699         AYLESTONE AVENUE  2.286667e+06  51.540916   -0.217874
853          BARONSMEAD ROAD  2.375000e+06  51.477315   -0.239457
979           BEAUCLERC ROAD  2.480000e+06  51.499577   -0.229033
1100         BELVEDERE DRIVE  2.340000e+06  51.424917   -0.212077
1213       BICKENHALL STREET  2.208500e+06  51.521197   -0.158934
1251       BIRCHLANDS AVENUE  2.217000e+06  51.448394   -0.160468
1550          BRAMPTON GROVE  2.456875e+06  51.570365   -0.283394
1629       BRIARDALE GARDENS  2.397132e+06  51.560175   -0.195431
1794                BROOKWAY  2.400000e+06  45.432185 -122.802812
1911            BURBAGE ROAD  2.445000e+06  51.450967   -0.095288
1977               BURY WALK  2.492500e+06  52.145529   -0.423593
2064         CALLCOTT STREET  2.375000e+06  51.508350   -0.198328
2125       CAMPDEN HILL ROAD  2.379653e+06  51.506461   -0.198896
2132            CAMPION ROAD  2.461000e+06  52.681351    0.965439
2155           CANNING PLACE  2.425000e+06  53.402201   -2.987260
2222           CARLISLE ROAD  2.200000e+06  42.549099  -71.416669
2227         CARLTON GARDENS  2.483500e+06 -37.801943  144.971970
2239           CARLYLE COURT  2.300000e+06  41.119072  -75.061516
2402          CHALCOT SQUARE  2.286679e+06  51.541455   -0.155265
2480            CHARLES LANE  2.414000e+06  40.237525  -83.349229
2558        CHELSEA CRESCENT  2.495000e+06  34.524738  -85.448539
2602     CHESTER CLOSE NORTH  2.450000e+06  51.529205   -0.145081
2633            CHEYNE COURT  2.250000e+06  51.599677    0.525623
2636              CHEYNE ROW  2.410000e+06  51.483717   -0.169603
2681           CHISWICK MALL  2.287500e+06  51.487994   -0.246605
2756               CITY ROAD  2.468340e+06  51.528306   -0.091946
...                      ...           ...        ...         ...
10896   RUSSELL GARDENS MEWS  2.300000e+06  51.499869   -0.211890
11143         SETTLES STREET  2.487500e+06  51.515324   -0.064327
11218         SHELDON AVENUE  2.349542e+06  43.599287  -73.045426
11451          SOUTH END ROW  2.470000e+06  51.498746   -0.189079
11529    SOUTHWOOD LAWN ROAD  2.350000e+06  51.574627   -0.146238
11532         SOVEREIGN PARK  2.500000e+06 -27.493709  153.210145
11749  ST MARGARETS CRESCENT  2.216500e+06  53.607492   -0.267971
11785       ST OSWALDS PLACE  2.250000e+06  51.487207   -0.118534
11805       ST PETERS SQUARE  2.468730e+06  41.902235   12.457839
11842       STAFFORD TERRACE  2.355000e+06  51.500938   -0.196049
12180       SUTHERLAND PLACE  2.456000e+06  54.773179   -1.543816
12242          SYDNEY STREET  2.240833e+06 -33.805144  151.193327
12384            THAMES BANK  2.400000e+06  51.471715   -0.268907
12446            THE HEXAGON  2.335000e+06 -31.895673   26.875812
12725        TREDEGAR SQUARE  2.436666e+06  51.527093   -0.032184
12787         TRINITY STREET  2.317500e+06  52.206141    0.118133
12937   UPPER HAMPSTEAD WALK  2.500000e+06  51.558467   -0.177453
13198        WALPOLE GARDENS  2.303500e+06  51.439743   -0.341263
13200         WALPOLE STREET  2.242500e+06  52.626946    1.284655
13274         WARWICK SQUARE  2.432273e+06  33.738563 -117.846506
13371            WELBECK WAY  2.267000e+06  51.517119   -0.148458
13383      WELLESLEY TERRACE  2.410000e+06  51.529214   -0.092933
13394      WELLINGTON STREET  2.293155e+06  22.281077  114.156045
13526     WESTMORELAND PLACE  2.300000e+06  40.740224 -111.846323
13633       WHITFIELD STREET  2.451000e+06  53.492018   -2.242218
13678         WILFRED STREET  2.410538e+06  53.511619   -2.194728
13704     WILLOW BRIDGE ROAD  2.425000e+06  51.543109   -0.095508
13724          WILSON STREET  2.257500e+06  38.439464 -122.722196
13752        WINCHENDON ROAD  2.350000e+06  51.476958   -0.203

**iv. Modeling**

In [24]:
address = 'London, UK'

geolocator = Nominatim()
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print('The geograpical coordinate of London City are {}, {}.'.format(latitude, longitude))

/opt/conda/envs/Python36/lib/python3.6/site-packages/ipykernel/__main__.py:3: DeprecationWarning: Using Nominatim with the default "geopy/1.20.0" `user_agent` is strongly discouraged, as it violates Nominatim's ToS https://operations.osmfoundation.org/policies/nominatim/ and may possibly cause 403 and 429 HTTP errors. Please specify a custom `user_agent` with `Nominatim(user_agent="my-application")` or by overriding the default `user_agent`: `geopy.geocoders.options.default_user_agent = "my-application"`. In geopy 2.0 this will become an exception.
  app.launch_new_instance()


The geograpical coordinate of London City are 51.4893335, -0.144055084527687.


In [26]:
# create map of London using latitude and longitude values
map_london = folium.Map(location=[latitude, longitude], zoom_start=11)

# add markers to map
for lat, lng, price, street in zip(df['Latitude'], df['Longitude'], df['Avg_Price'], df['Street']):
    label = '{}, {}'.format(street, price)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_london)  
    
map_london